Microstrain px4 driver filter

JoelJ18 opened the pull request.

bkueng committed:
rpm_capture: add rpm capture driver, publising pwm_input (for now)

AlexKlimaj committed:
ARK TESEO Bootloader Placeholder

MaEtUgR committed:
mavlink_receiver: don't publish out of range joystick input

Note that the MAVLink definition explicitly writes
"A value of INT16_MAX indicates that this axis is invalid."
which before this change was happily executed.

bresch committed:
AlphaFilter: allow setting dt in update call

bresch committed:
AlphaFilter: set time constant instead of alpha

Then the update function can set the dt at every iteration if needed

MaEtUgR committed:
control_allocator: add unit for slew rate and reword description

GuillaumeLaine committed:
msg: introduce message versioning

GuillaumeLaine committed:
tools: update scripts after msg/ restructure

GuillaumeLaine committed:
jenkins: update after msg/ restructure

sfuhrer committed:
Navigator: RTL_direct: onyl start precision land if param is set to enable

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

Claudio-Chies committed:
SENS: RNG: SF45 changed data processing and publication design, moved to a publishing per sector design.
other minor improvements

MaEtUgR committed:
control_allocator: fix typo and use [0,1] instead of [0%, 100%] in slew rate description

Co-authored-by: Hamish Willee <hamishwillee@gmail.com>

chfriedrich98 committed:
differential: adjust speed setpoint based on yaw rate setpoint

sfuhrer committed:
rtl_direct: don't check RTL_PLD_MD param twice but use _mission_item.land_precision

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

sfuhrer committed:
RTL params: make clear that RTL_PLD_MD does not apply for mission RTL destinations

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

sfuhrer committed:
RTL fast reverse: enable precision landing

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

sfuhrer committed:
MissionBase: remove unused param

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

chfriedrich98 committed:
mecanum: deprecate RM_MISS_SPD_DEF

chfriedrich98 committed:
mecanum: deprecate RM_MAN_YAW_SCALE

chfriedrich98 committed:
mecanum: add slew rates to yaw, yaw rate and speed setpoints

MaEtUgR committed:
Add new C++ PID library

MaEtUgR committed:
Replace old pid library with new one

haumarco committed:
SIH: ellipsoidal earth model

SIH: use projection functions and constants from geo lib

SIH: remove unnecessary member variable

SIH: clarify names of rotation matrices and frames

SIH: do not store DCM corresponding to quaternion attitude

Using DCM is more efficient when more than 1 rotation needs to be done,
which is not the case here.

SIH: don't store local variable as member

SIH: use Wgs84 constants everywhere

SIH: do not store delta_quaternion

Converting an AxisAngle to a Quaternion uses the exponenial

SIH: organise ECEF member variables

SIH: add earth spin rate to gyro data

Co-authored-by: bresch <brescianimathieu@gmail.com>

StefanoColli committed:
Navigator: add optional delay after gimbal mission items

perrrre committed:
Adding tiltrotor model

dagar committed:
ekf2: derivation.py remove sideslip small angle approximation

julianoes committed:
drivers: Copy SPL06 to SPA06

julianoes committed:
drivers: add SPA06

MaEtUgR committed:
mission_block: fix style, shorten debug message strings

MaEtUgR committed:
mission_block: simplify timeout check

chfriedrich98 committed:
mecanum: fix inverse kinematics

chfriedrich98 committed:
mecanum: adjust speed setpoints to always be feasible

chfriedrich98 committed:
mecanum: update SITL airframe parameters

chfriedrich98 committed:
differential: add slow down effect in mission mode

MaEtUgR committed:
navigator: unify timeout waiting for payload to execute mission item command

Used for winch, gripper, gimbal to reach the desired state before continuing the mission.
Ideally we'd have feedback from all these components and not just a feed-forward delay.

julianoes committed:
boards: add SPA06 to KakuteH7/H7mini/H7v2 boards

jnippula committed:
Logger: combined key & cipher file

Use one single .ulge file to store both wrapped symmetric key and
encrypted ulog data instead of creating separate .ulgk/ulgc files

jnippula committed:
Tools/decrypt_ulog.py: support for .ulge log files

Claudio-Chies committed:
Included turning radius calculations for vertical changes and removed the requirement for the same altitude in the 2D turning radius logic.

Claudio-Chies committed:
MPC: PositonSmoothing, change test to reflect that we have to come inwithing the acceptance radius, and not exact position.

alexcekay committed:
SENS: RNG: SF45: Fixed sf45 parser, added general checks to avoid potential out-of-bound access

GuillaumeLaine committed:
uxrce_dds_client: remove unused msgdir arg from script

sfuhrer committed:
COMMON_DISTANCE_SENSOR: remove leddar_one from the list

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

sfuhrer committed:
COMMON_DISTANCE_SENSOR: remove gy_us42 from the list

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

sfuhrer committed:
COMMON_DISTANCE_SENSOR: remove srf02 from the list

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

GuillaumeLaine committed:
docs: split msgs as un/versioned

bkueng committed:
standard_modes: add vehicle-type specific standard modes

See https://mavlink.io/en/messages/development.html#MAV_STANDARD_MODE.
The only standard mode that is not set is MAV_STANDARD_MODE_SAFE_RECOVERY,
as PX4 uses RTL for that (with configuration parameters).

mrpollo committed:
ci: tag container main & main-date

Make sure we always have a { branch name } container tag in addition to
a { branch name + date } tag. This way we have a rolling { branch name }
of the main and release branches

mrpollo committed:
ci: push px4-dev container to docker hub registry

mrpollo committed:
ci: publish pr images to registry

bresch committed:
SIH: refactor MC ground contact

bresch committed:
SIH: rework FW ground contact

mrpollo committed:
readme: update ci badge

bresch committed:
SIH: do not assume being a tailsitter when creating airspeed measurement

bresch committed:
SIH-FW: fix aileron and elevator signs

This broken when changing from mixer files to the control allocation module.

bresch committed:
SIH-FW: allow pitching up during takeoff

Otherwise difficult to get lift

mrpollo committed:
ci: disable publishing PR images to docker hub

Docker hub is rate limiting our API access, as a result tests are
failing for no apparent reason. This change will decrease the API calls
by at least 80%

We have applied for an Open Source account with greater API limits, I
will come back to this and update as necessary when and if they grant us
access to their program.

mrpollo committed:
ci: update actions and images

Updates some actions to run build steps in container instead of the
whole workflow.

bresch committed:
matrix: allow casting float<->double

bresch committed:
lla: add functions to convert from and to ECEF

bresch committed:
lla: move to lib directory

bresch committed:
lla: add gravity constant at equator

bresch committed:
SIH: use LatLonAlt class

bresch committed:
SIH: add transport rate acceleration to local acceleration

bresch committed:
SIH: set GNSS delay to 0 as delay is not simulated

bresch committed:
SIH: lower IMU noise before takeoff

This speeds-up the EKF alignment

MaEtUgR committed:
PID: protect from division by zero because of dt

Co-authored-by: chfriedrich98 <125505139+chfriedrich98@users.noreply.github.com>

MaEtUgR committed:
PID: Fix test to respect integral updates being applied in the next iteration

Co-authored-by: chfriedrich98 <125505139+chfriedrich98@users.noreply.github.com>

chfriedrich98 committed:
mecanum: update current position in main file

MaEtUgR committed:
RPMCapture: add hardcoded rpm processing

Such that we can continue development on this part.
The implementation was already used before porting things into the RPM capture module.

bresch committed:
SIH-plane: fix actuator mapping

MaEtUgR committed:
RPMCapture: Detect rpm sensing timeout

MaEtUgR committed:
RPMCapture: switch to PublicationMulti to be compatible with the existing rpm drivers

MaEtUgR committed:
RPM: clean up message

MaEtUgR committed:
RPMCapture: much better timeout scheduling

Not timing out based on a random interval but based
on the time after the last inerrupt.

MaEtUgR committed:
mavlink: stream RAW_RPM message for ONBOAR_LOW_BANDWIDTH

mrpollo committed:
container: use PX4 tags whiel tagging images

sfuhrer committed:
RPM capture: publish every 1s without new data

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

dagar committed:
Tools/astyle: check_code_style_all.sh skip pre-commit hook if non-interactive

dagar committed:
uxrce_dds_client: update cmake requirements to match Micro-XRCE-DDS-Client submodule

hamishwillee committed:
airframes markdownout.py - br rather than p for generated code

bresch committed:
EKF2: reset aid src when resetting state to measurement

Filtered innovations and test ratios can be large before the reset and
would trigger pre-flight warnings

bresch committed:
add SensorAgpSim to generate Aux Global Position data

MaEtUgR committed:
TrajectoryConstraints: clarify waypoint indexing

MaEtUgR committed:
PositionSmoothing: refactor _getMaxZSpeed()

haumarco committed:
fix sih hitl plane airframe

slgrobotics committed:
boards: emlid_navio - support 64-bit OS on Raspberry Pi 4,5 (#24006)

adds the `emlid_navio2_arm64` build target - supports 64-bit OS on Raspberry Pi 4,5 (#24006)

bresch committed:
RpmControl: remove raw RPM spikes using median filter

Claudio-Chies committed:
Added possibility to modify the start position from external sources, as its done in the _getMaxXYSpeed

bresch committed:
disable SIH on x21-777 and v6u to save flash

chfriedrich98 committed:
mecanum: update parameter description

MaEtUgR committed:
Commander: remove `COM_POS_FS_DELAY`

A user configurable delay for the internal `vehicle_local_position` seems confusing in my eyes. It's a different timeout for fixed-wing and multirotor which might have made sense earlier but not really anymore since the topic is constantly published by the estimator and not expected to time out on either vehicle type and the parameter description is also misleading because it's outdated.

perrrre committed:
Add gz model for quadtailsitter (#23943)

* Add gazebo airspeed plugin and add a tailsitter model
---------

Co-authored-by: Claudio Chies <61051109+Claudio-Chies@users.noreply.github.com>

bresch committed:
lla; fix conversion to ECEF

bresch committed:
EKF2: reset global position using variance

alexcekay committed:
SENS: RNG: SF45:Fix startup problems, increase frequency, robust parser, use nonblocking reads

perrrre committed:
ESC check: Avoid unsigned timestamp underflow in telemtry timeout (#24069)

* Avoid unsigned integer underflow

* ESC check: add brackets to timeout for readability

---------

Co-authored-by: Matthias Grob <maetugr@gmail.com>

perrrre committed:
Remove gz-garden as it reached EOL

Claudio-Chies committed:
SENS: RNG: SF45: Added timeout to sensor measurements, to compensate the lower loop time of CollisionPrevention

MaEtUgR committed:
offboardCheck: acceleration setpoints only require vehicle attitude control

julianoes committed:
gps: heading fixes for NMEA/Unicore

This updates the GPS submodule which includes NMEA/Unicore fixes:

- Add correct return value for sat infos
- Only publish on position updates
- Request required topics at 5 Hz for Unicore

dakejahl committed:
params: flash: erase corrupt sector (#24065)

PX4BuildBot committed:
update all px4board kconfig

Jaeyoung-Lim committed:
Add mavlink stream for plane camera (#24081)

bresch committed:
ekf2_replay: log groundtruth messages

bresch committed:
SIH-AGP: add sensor failure simulation

riccardo-mengoli committed:
Mission: Replay gimbal cached items before reaching mission waypoint (#24085)

When flying patterns, photos are sometimes taken while the gimbal is pitching up or down. To address this, we orient the gimbal before reaching the mission waypoint, allowing more time to complete the action. Additionally, we verify if the vehicle is climbing to avoid orienting the gimbal while on the ground.

alexcekay committed:
Added various features to flash analysis (#24072)

* Added various features to flash analysis (also run with FLASH overflow, summary in comment output, newer bloaty version, only add comment if change is large enough, ...)

* Added feedback from review

* Use wildcards

* Removed backward-compat logic and use correct base

alexcekay committed:
Also consider targets that include a '-' (#24087)

oravla5 committed:
RPMCapture: added pulse per rev parameter. Do not publish if pulses are to fast

MaEtUgR committed:
RPMCapture: fix pulses per revolution parameter name, description, type -> use module params

MaEtUgR committed:
RPMCapture: refactor to clarify when an interrupt happened

MaEtUgR committed:
RPMCapture: refactor to clarify scheduling

MaEtUgR committed:
RPMCapture: define period as UINT32_MAX when timed out

MaEtUgR committed:
RPMCapture: publish outliers in rpm message but don't update the filters and estimate

MaEtUgR committed:
RPMCapture: add back filter reset on timeout

MaEtUgR committed:
RPMCapture: fix dt should not have a minimum which would limit to 6000rpm

vincentpoont2 committed:
update H-Flow Boot config and add stm32_configgpio (#24086)

julianoes committed:
listener: only clear screen with multiple messages (#24019)

* listener: only clear screen with multiple messages

* listener: fully clear before printing again

Otherwise, we end up seeing artifacts from the previous print.

bresch committed:
ekf2: set baro bias when GNSS is alt ref

Do this even when GNSS altitude fusion is disabled.

oravla5 committed:
RpmControl: class clean up

MaEtUgR committed:
Airframes: remove unnecessary double newlines

MaEtUgR committed:
PCF8583: refactor 1e6f for 1000000.f

MaEtUgR committed:
LidarLitePWM: refactor static cast and comment

MaEtUgR committed:
ControlAllocator: introduce helicopter rotor rpm controller

MaEtUgR committed:
RpmControl: name current timestamp now following the convention

MaEtUgR committed:
RpmControl: call local message instance after message name following the convention

mrpollo committed:
container: remove pr container push to registry

MaEtUgR committed:
RpmControl: remove status message because it by now only contains redundant information

bkueng committed:
mavlink: update submodule

MAV_STANDARD_MODE_SAFE_RECOVERY and MAV_STANDARD_MODE_RETURN_HOME got
merged.

MaEtUgR committed:
RPMCapture: fix copyright year, improve comments

MaEtUgR committed:
camera_capture: refactor to simplify capture channel initialization

MaEtUgR committed:
Use defines instead of hardcoded magic values for maximum number of output channels and rpm filter time constant

bresch committed:
ekf-agp: fix timeout

MaEtUgR committed:
Add IO timer channel mode for RPM and also sync up rpi configuration to work with all these drivers

dakejahl committed:
sim: gz: remove Garden from cmake

sfuhrer committed:
mixer: FunctionMotors: leave NAN control values at NAN with non-zero THR_MDL_FAC

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

sfuhrer committed:
mixer: FunctionMotors: compare against FLT_EPSILON instead of 0.f

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

PX4BuildBot committed:
Update submodule mavlink to latest Sat Dec 14 00:39:04 UTC 2024

    - mavlink in PX4/Firmware (661ae0c0e3ac493ceddd13120e8ccb5bac47d887): https://github.com/mavlink/mavlink/commit/0e420102dbdd7e9f59617dd00c05a2470f22eef2
    - mavlink current upstream: https://github.com/mavlink/mavlink/commit/5e3a42b8f3f53038f2779f9f69bd64767b913bb8
    - Changes: https://github.com/mavlink/mavlink/compare/0e420102dbdd7e9f59617dd00c05a2470f22eef2...5e3a42b8f3f53038f2779f9f69bd64767b913bb8

    5e3a42b8 2024-12-12 Hamish Willee - development: Remove parameter transaction proposal (#2169)
25f4e9f0 2024-12-11 Hamish Willee - STANDARD_MODE_SAFE_RECOVERY + RTL merge (#2191)
d5a8cb43 2024-12-11 Hamish Willee - Docs generate fix (#2194)
35f70c4a 2024-12-05 Hamish Willee - Create index.md, dialects.md and make notes/warning work with vitepress (#2193)
57c02856 2024-12-05 Hamish Willee - CAMERA_THERMAL_RANGE - no longer WIP (#2189)
1ee2ebe1 2024-12-04 Hamish Willee - fetch_dialect_ardupilotmega.yml: Not fail if nothing to commit (#2181)
1aa8c2d0 2024-12-04 Hamish Willee - ardupilotmega - remove_mav_cmd_external_estimate (#2187)
49fa509a 2024-11-30 github-actions[bot] - ardupilotmega dialects from ArduPilot/mavlink: Fri Nov 29 23:49:36 UTC 2024 (#2185)
75ebfc8f 2024-11-30 Hamish Willee - Update fetch_dialect_ardupilotmega.yml - fix copy error (#2184)
8e97709d 2024-11-30 Hamish Willee - fetch_dialect_ardupilotmega.yml - add loweheiser (#2183)
82b81aa8 2024-11-27 github-actions[bot] - ardupilotmega.xml from ArduPilot/mavlink: Wed Nov 27 04:13:16 UTC 2024 (#2178)
5c421a33 2024-11-27 Hamish Willee - Speed uncertainty units /s (#2180)
33f8a327 2024-11-24 David Sastre - Some modifications to multi-GCS protocol (#2179)
9938f940 2024-11-21 Hamish Willee - fetch_dialect_ardupilotmega - create (#2177)
a55d0ec5 2024-11-21 Hamish Willee - MAV_CMD_GROUP_START and _GROUP_END delete wip (#2176)
1cf3c721 2024-11-21 Hamish Willee - dev: move FUEL_STATUS to common (#2170)
7ecb8e33 2024-11-21 Hamish Willee - WIFI_NETWORK_SECURITY - wip removal (#2164)

bresch committed:
ekf2: use alpha filter class

bresch committed:
ekf: update change indicator

The calculation of "alpha" in the accel_horiz and height_rate lowpass
filters is slightly different than before.

RomanBapst committed:
Fixed-wing Position Control: Split up control methods for VTOL backtransition (#23893)

* split methods to control backtransition depening on availablity of position

Signed-off-by: RomanBapst <bapstroman@gmail.com>

* fixed sign error and replace hardcoded number with constant

Signed-off-by: RomanBapst <bapstroman@gmail.com>

* make changes such that controller holds initial heading during transition

Signed-off-by: RomanBapst <bapstroman@gmail.com>

* use reference instead of copy

Signed-off-by: RomanBapst <bapstroman@gmail.com>

* added comment

Signed-off-by: RomanBapst <bapstroman@gmail.com>

* flash reduction

Signed-off-by: RomanBapst <bapstroman@gmail.com>

---------

Signed-off-by: RomanBapst <bapstroman@gmail.com>

bresch committed:
ekf2: do not pre-compute airspeed Kalman gain

The generated code is not much faster than the simple matrix-vector
multiplication

MaEtUgR committed:
RpmControl: maximum rpm outliers are now caught by RpmCapture

MaEtUgR committed:
RpmControl: split into cpp source file fixing includes

MaEtUgR committed:
RpmControl: simplify the entire control logic

dagar committed:
commander: accelerometer calibration respect rotation

 - accel cal use Accelerometer calibration class to fully respect
   rotation (both internal and external sensors)

bresch committed:
yaw_est: store attitude as quaternion instead of DCM

This saves flash and makes code simpler

Jaeyoung-Lim committed:
Dynamically update ff gain for fw rate control (#24120)

MaEtUgR committed:
Commander: start timer for auto disarm after spoolup

MaEtUgR committed:
Helicopter defaults: don't auto disarm so quickly after spoolup

MaEtUgR committed:
RpmControl: Better consider the case where there's no rpm measurement (anymore)

julianoes committed:
gimbal: correctly set gimbal_device_id

When we use a gimbal connected via "RC", so PWM via the Aux channels, we
need to set the gimbal_device_id to 1 as per the protocol.

This was missing for GIMBAL_DEVICE_ATTITUDE_STATUS, so I added that, and
fixed the name of that variable while at it.

dirksavage88 committed:
uxrce_dds_client: include distance sensor in dds_topics.yaml

Signed-off-by: dirksavage88 <dirksavage88@gmail.com>

StefanoColli committed:
GZ: add gimbal simulation (#23382)

haumarco committed:
dont use BROACAST_MODE_ON during unit testing

sfuhrer committed:
ADSB: pass now timestamp as arguemnt of get_traffic_state()

That enables us to time the unit tests better plus saves some flash and CPU.

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

sfuhrer committed:
ADSB: pass now timestamp as arguemnt of send_traffic_warning()

Saves some flash and CPU by not having to call hrt_absolute_time().

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

MaEtUgR committed:
FlightTaskTransition: clean up and simplify

MaEtUgR committed:
FlightTaskTransition: keep high decelration when overshooting the transition target

perrrre committed:
Adding min-max angles to use for normalized servo outputs

perrrre committed:
Update other airframes with min max values different than default

perrrre committed:
Adding gimbal rate in gz simulation (#24125)

* Adding gimbal rate in gz simulation

* add submodule

sfuhrer committed:
ADSB: improve readability of unit tests

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

MaEtUgR committed:
gz submodule: switch to hash on submodule main branch

after https://github.com/PX4/PX4-gazebo-models/pull/71 was merged

MaEtUgR committed:
ADSB: replace last excess hrt_absolute_time() in the loop

MaEtUgR committed:
mc_pos_control: shorten parameter descriptions to a readable size

dakejahl committed:
boards: ark_fpv: add msp osd

MaEtUgR committed:
Navigator: make ADSB handling optional

sfuhrer committed:
update gz submodule to get new tiltrotor sim

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

MaEtUgR committed:
FlightTaskTransition: Comment why using invalid horizontal velocity works

sfuhrer committed:
BatterySimulation: remove unused _battery_pub

The simulated battery is published by the regular battery class.

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

mrpollo committed:
ci: move to px4 git tag versions

sfuhrer committed:
Simulation: add SystemPowerSimulator to publish SystemPower.msg in sim

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

sfuhrer committed:
SITL: remove setting CBRK_SUPPLY_CHK by default

This allows us to run battery failsafe tests again in SITL
wihtout having to change params first.

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

RomanBapst committed:
Navigator: Set altitude acceptance radius to infinity when moving to land point after transition (#24115)

* Navigator: set alt acceptance radius to infinity
for land waypoint after backtransition -> avoid
vehicle with depleted battery from not reaching the alt
setpoint and getting stuck

Signed-off-by: RomanBapst <bapstroman@gmail.com>

mahimayoga committed:
Collision Prevention: Scale obstacle distance with vehicle attitude for varying sensor orientations  (#24107)

bresch committed:
ekf2: fix GNSS drift false alarm

In SIH, the GNSS signal is zero-mean, but apparently not symmetric. The
issue is that saturating such a signal creates an artificial bias. This
made the check fail as the bias was above the threshold.

Commit:
boards: add micoair743-v2 config files to PX4-Autopilot

MaEtUgR committed:
Carzyflie: don't compile Navigator ADSB handling

I forgot to stage that change in the original change:
839010eeab8b0ff6f472d102d89d367eddcbaff9

Commit:
boards: fix some configs for micoair743 and aio

dagar committed:
uavcan: relax UAVCAN_RNG_MIN/UAVCAN_RNG_MAX defaults (#24140)

- this is optional sensor metadata that can be set if actually needed

sfuhrer committed:
Commander: remove not used include of blocks.hpp

Signed-off-by: Silvan <silvan@auterion.com>

dakejahl committed:
test: add test for range finder fusion while on ground

bresch committed:
ekf2-rng: do not continuously reset terrain in rng height ref

When the height reference datum is ground level (range finder height
ref), the terrain state is not updated through fusion and should stay at
a constant altitude.

bresch committed:
fw-att: fix wheel controller

NicolasM0 committed:
local position acceleration: use mean value between two publications (#24105)

To avoid aliasing on the ned acceleration, add an accumulation of acceleration to improve the downsampling

bresch committed:
fw-att: avoid multiple conversions

jnippula committed:
logger: fix file open issue if crypto algorithm is disabled

move init_logfile_encryption() call after the buffer start_log() call
to have log file already open while storing the header and key data to
the beginning of the file.

AlexKlimaj committed:
analog_battery: add option for filter (#23987)

AlexKlimaj committed:
boards: ark flow set SENS_FLOW_RATE default to 150 (#24170)

ZeroOne-Aero committed:
ZeroOne X6: add IST8310 support

bresch committed:
ekf2: fix comments about primary height sources

Having the xxx_hgt flag set only means that the sensor is fused for
altitude/height aiding. It's not necessarily the height reference.

RomanBapst committed:
remove fixed-wing condition from airspeed fusion condition
- enables airspeed fusion to begin during transition already
- airspeed fusion is started based on airspeed being above defined threshold

Signed-off-by: RomanBapst <bapstroman@gmail.com>

robbiefish commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -70,6 +70,50 @@ MicroStrain::MicroStrain(const char *uart_port) :
+		param_set(param_find("EKF2_EN"), &m);
+
+		m = 0;
+		param_set(param_find("SENS_IMU_MODE"), &m);
```
Why does setting this and the next parameter to 0 disable EKF2? If you know, might be good to point that out, otherwise it is fine as is

robbiefish commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -292,6 +343,7 @@ mip_cmd_result MicroStrain::configureImuMessageFormat()
 {
 	uint8_t num_imu_descriptors = 0;
 	mip_descriptor_rate imu_descriptors[4];
+	PX4_INFO("Start2 %d", sizeof(imu_descriptors));
```
This could probably be changed to a debug log, and made more descriptive or be removed

robbiefish commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -329,6 +381,7 @@ mip_cmd_result MicroStrain::configureImuMessageFormat()
 		imu_descriptors[num_imu_descriptors++] = mip_descriptor_rate { MIP_DATA_DESC_SENSOR_PRESSURE_SCALED, baro_decimation};
 	}
 
+	PX4_INFO("End2 %d", sizeof(imu_descriptors));
```
Same here. Could probably be made debug or removed

robbiefish commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -338,6 +391,89 @@ mip_cmd_result MicroStrain::configureImuMessageFormat()
+	uint8_t num_filter_descriptors = 0;
+	mip_descriptor_rate filter_descriptors[11];
+
+	PX4_INFO("Start %d", sizeof(filter_descriptors));
```
Debug log

robbiefish commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -338,6 +391,89 @@ mip_cmd_result MicroStrain::configureImuMessageFormat()
+		filter_descriptors[num_filter_descriptors++] = mip_descriptor_rate { MIP_DATA_DESC_FILTER_FILTER_STATUS, filter_decimation_low};
+	}
+
+	PX4_INFO("Hi: %d, %d, %d, %d", num_filter_descriptors, filter_decimation_high, filter_decimation_low,
```
Debug log

robbiefish commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -349,6 +485,139 @@ mip_cmd_result MicroStrain::writeBaudRate(uint32_t baudrate, uint8_t port)
+	}
+
+	else {
+		res = mip_filter_write_aiding_measurement_enable(&_device,
```
Some devices may not support the aiding measurement at all. If this is the case, and the user did not request it, it shouldn't be an error, just maybe log a debug log

robbiefish commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -349,6 +485,139 @@ mip_cmd_result MicroStrain::writeBaudRate(uint32_t baudrate, uint8_t port)
+	mip_cmd_result res;
+
+	// Selectively turn on the mag aiding source
+	if (_param_cv7_int_mag_en.get() == 1) {
```
Enabling and disabling aiding measurements could probably be made into a convenience function as it is pretty much the same thing with different parameters, and it would cut down on code considerably

drobb257 replied:
1. Make sure that you check the device supports the command before sending.  
2. Raise an error if the device returns any error on a user enabled aiding source

robbiefish commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -349,6 +485,139 @@ mip_cmd_result MicroStrain::writeBaudRate(uint32_t baudrate, uint8_t port)
+	}
+
+	// Sets up multi antenna offsets for the GNSS/INS if its supported
+	if (supportsDescriptorSet(MIP_GNSS1_DATA_DESC_SET)) {
```
This check and comment are confusing. It looks like what you are trying to do here is to check if you should configure an aiding frame, or configure antenna offsets. I think that check needs to be more complex.

In pseudocode, it should be something like:

```
if (supports_gnss1_descriptor_set && supports_multi_antenna_offset)
{
  // configure GNSS antenna offsets
}
else if (supports_aiding_frame_config)
{
  // configure aiding frame for GNSS
}
```

robbiefish commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -349,6 +485,139 @@ mip_cmd_result MicroStrain::writeBaudRate(uint32_t baudrate, uint8_t port)
+		// Enables dual antenna heading as an aiding measurement
+		if (!_param_ms_heading_en.get()) {
+			res = mip_filter_write_aiding_measurement_enable(&_device,
+					MIP_FILTER_AIDING_MEASUREMENT_ENABLE_COMMAND_AIDING_SOURCE_EXTERNAL_HEADING, true);
```
I think this should be `MIP_FILTER_AIDING_MEASUREMENT_ENABLE_COMMAND_AIDING_SOURCE_GNSS_HEADING`, and line 521 should be `MIP_FILTER_AIDING_MEASUREMENT_ENABLE_COMMAND_AIDING_SOURCE_EXTERNAL_HEADING`

drobb257 replied:
Do we want this to enable dual antenna?  Right now it configures external heading aiding, not internal dual antenna

robbiefish commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -349,6 +485,139 @@ mip_cmd_result MicroStrain::writeBaudRate(uint32_t baudrate, uint8_t port)
+	PX4_INFO("CV7");
+
+	// Otherwise sets up the aiding frame for the INS
+	return mip_aiding_write_frame_config(&_device, 2, MIP_AIDING_FRAME_CONFIG_COMMAND_FORMAT_EULER, false,
```
I would put this inside an else if like noted in the above comment, and add comments to note that we will prioritize setting antenna offsets if we can, but will otherwise set aiding frame config

robbiefish commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -349,6 +485,139 @@ mip_cmd_result MicroStrain::writeBaudRate(uint32_t baudrate, uint8_t port)
+	}
+
+	// Filter initialization configuration
+	return mip_filter_write_initialization_configuration(&_device, 0,
```
Should check if the device supports this command and log if not

robbiefish commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -349,6 +485,139 @@ mip_cmd_result MicroStrain::writeBaudRate(uint32_t baudrate, uint8_t port)
+	_gps_origin_ep[0] = gp.eph;
+	_gps_origin_ep[1] = gp.epv;
+
+	return mip_filter_write_rel_pos_configuration(&_device, 1, 2, _ref_pos);
```
Same here. Check if the device supports the command, and log if not

robbiefish commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -369,6 +638,12 @@ bool MicroStrain::initializeIns()
 	}
 
+	// Get the supported descriptors for the device in use
+	if (getSupportedDescriptors() != MIP_ACK_OK) {
```
This and all other configuration (not having to do with baudrate) should happen after setting the baudrate to the desired value. The log after this makes me think that this is happening before that

drobb257 commented on src/drivers/ins/microstrain/MicroStrain.hpp:
```diff
@@ -134,23 +161,70 @@ class MicroStrain : public ModuleBase<MicroStrain>, public ModuleParams, public
+		(ParamInt<px4::params::MS_HEADING_EN>) _param_ms_heading_en
 	)
 
+	uORB::Publication<vehicle_global_position_s> _vehicle_global_position_pub{(_param_ms_mode.get() == 0) ? ORB_ID(external_ins_global_position) : ORB_ID(
```
Does this make sense to put this code in the header file?  Or in the constructor of the source file?

robbiefish commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -402,20 +671,68 @@ bool MicroStrain::initializeIns()
+					       &filterCallback,
+					       this);
+
+	PX4_INFO("PARAMS %f/%f/%f / %f/%f/%f / %f/%f/%f", (double)gnss_antenna_offset1[0], (double)gnss_antenna_offset1[1],
```
I like logging the parameters. I would suggest moving this to wherever you read the parameters and print them with their names so it is easy to see what is getting set

robbiefish commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -402,20 +671,68 @@ bool MicroStrain::initializeIns()
+	}
+
+	// Reset the filter, then set the initial conditions
+	if (mip_filter_reset(&_device) != MIP_ACK_OK) {
```
I think that resetting the filter should be one of the last steps of configuration, and definitely after filter configuration

drobb257 commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -472,35 +784,391 @@ void MicroStrain::sensorCallback(void *user, const mip_packet *packet, mip::Time
+	}
+
+	// Publish only if the required data was extracted from the packet
+	bool vgp_valid = pos_llh.updated && llh_uncert.updated;
```
Write out variable description: vehicle_global_position_valid rather than vgp_valid

robbiefish commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -472,35 +784,391 @@ void MicroStrain::sensorCallback(void *user, const mip_packet *packet, mip::Time
+
+	}
+
+	if (vo_valid && ref->_param_ms_mode.get() == 1) {
```
This might be fine, but there may be overhead from refetching parameters. It may be better to get this at config time, store it in a member variable and check that instead

robbiefish commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -472,35 +784,391 @@ void MicroStrain::sensorCallback(void *user, const mip_packet *packet, mip::Time
+		ref->_vehicle_odometry_pub.publish(vo);
+	}
+
+	if (vav_valid && ref->_param_ms_mode.get() == 1) {
```
Same here. Make sure this does not perform network or disk operations to fetch this variable

drobb257 commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -338,6 +391,89 @@ mip_cmd_result MicroStrain::configureImuMessageFormat()
+
+	// Configure the Message Format depending on if the device supports the descriptor
+	uint16_t filter_decimation_high = base_rate / (uint16_t)_param_ms_filter_rate_high_hz.get();
+	uint16_t filter_decimation_low = base_rate / (uint16_t)_param_ms_filter_rate_low_hz.get();
```
Why are there two different decimation rate for the filter?

robbiefish commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -472,35 +784,391 @@ void MicroStrain::sensorCallback(void *user, const mip_packet *packet, mip::Time
+		ref->_vehicle_angular_velocity_pub.publish(av);
+	}
+
+	if (stat.updated && ref->_param_ms_mode.get() == 1) {
```
Same as above

drobb257 commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -338,6 +391,89 @@ mip_cmd_result MicroStrain::configureImuMessageFormat()
+
+	if (supportsDescriptor(MIP_FILTER_DATA_DESC_SET, MIP_DATA_DESC_FILTER_COMPENSATED_ANGULAR_RATE)
+	    && _param_ms_filter_rate_low_hz.get() > 0) {
+		filter_descriptors[num_filter_descriptors++] = mip_descriptor_rate { MIP_DATA_DESC_FILTER_COMPENSATED_ANGULAR_RATE, filter_decimation_low};
```
Why is the compensate angular rate output at a lower rate?

robbiefish commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -509,9 +1177,82 @@ bool MicroStrain::init()
+	}
+
+	mip_time t;
+	t.timebase = MIP_TIME_TIMEBASE_TIME_OF_ARRIVAL;
```
[SensorGps](https://docs.px4.io/main/en/msg_docs/SensorGps.html) has a `time_utc_usec` which you could probably use to get a more accurate timestamp, which could increase filter performance. Note that it is possible for this variable to be 0, so in that case you would probably have to not send it

robbiefish commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -509,9 +1177,82 @@ bool MicroStrain::init()
+	}
+
+
+// ---------------------------------------------------------- DEBUG START -----------------------------------------------------------------------
```
Should this be removed?

robbiefish commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -543,7 +1285,10 @@ void MicroStrain::Run()
 
 	mip_interface_update(&_device, false);
 
+	if (_param_ms_aiding_en.get() == 1) {sendAidingMeasurements();}
```
Again make sure it does not introduce overhead to get parameters at runtime

robbiefish commented on src/drivers/ins/microstrain/MicroStrain.hpp:
```diff
@@ -134,23 +161,70 @@ class MicroStrain : public ModuleBase<MicroStrain>, public ModuleParams, public
 	int _ms_schedule_rate_us{0};
 
+	template <typename T>
+	struct sensorSample {
```
This is good, but should be PascalCase, so `SensorSample`

robbiefish commented on src/drivers/ins/microstrain/module.yaml:
```diff
@@ -36,3 +36,134 @@ parameters:
+            type: int32
+            default: 1
+
+        MS_AIDING_EN:
```
I would break this out into different booleans for each position, velocity, and heading

robbiefish commented on src/drivers/ins/microstrain/module.yaml:
```diff
@@ -36,3 +36,134 @@ parameters:
+            description:
+                short: Enables Aiding Heading Measurements to the Sensor
+                long: |
+                    If enabled, the device will be configured to accept external heading as an aiding meaurement.
```
Instead of driving with only one variable, you should have something like `MS_EXT_HEADING_EN` for external heading, and `MS_DUAL_ANTENNA_EN`

drobb257 commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -349,6 +485,139 @@ mip_cmd_result MicroStrain::writeBaudRate(uint32_t baudrate, uint8_t port)
+{
+	sensor_gps_s gp{0};
+
+	while (true) {
```
Is this a loop that will just freeze the driver until a GPS fix is found?  If so, we might want to consider a different approach that doesn't sit locked in a busy loop.

drobb257 commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -349,6 +485,139 @@ mip_cmd_result MicroStrain::writeBaudRate(uint32_t baudrate, uint8_t port)
+		break;
+	}
+
+	_ref_pos[0] = gp.latitude_deg;
```
How does this get initialized in EKF2?  Is it off the first valid GNSS fix?  Or another method?

drobb257 commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -472,35 +784,391 @@ void MicroStrain::sensorCallback(void *user, const mip_packet *packet, mip::Time
+		vo.angular_velocity[1] = ang_rate.sample.gyro[1];
+		vo.angular_velocity[2] = ang_rate.sample.gyro[2];
+
+		vo.position_variance[0] = llh_uncert.sample.north;
```
Mixing uncertainty and variance?

drobb257 commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -472,35 +784,391 @@ void MicroStrain::sensorCallback(void *user, const mip_packet *packet, mip::Time
+		av.xyz[2] = ang_rate.sample.gyro[2];
+
+		// ------- Fields we cannot obtain -------
+		av.xyz_derivative[0] = 0;
```
Is this important?  We could always just numerically differentiate the compensated gyro signal

drobb257 commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -472,35 +784,391 @@ void MicroStrain::sensorCallback(void *user, const mip_packet *packet, mip::Time
+		vp.delta_vz = 0;
+		vp.vz_reset_counter = 0;
+
+		vp.unaided_heading = NAN;
```
Technically possible to populate, could do our own time integration

drobb257 commented on src/drivers/ins/microstrain/MicroStrain.cpp:
```diff
@@ -70,6 +70,50 @@ MicroStrain::MicroStrain(const char *uart_port) :
+	rotation.euler[2] = (float)_param_ms_sensor_yaw.get();
+
+	// Disables EKF2 to use the INS
+	if (_param_ms_mode.get() == 1) {
```
Let's not change external parameters automatically.  

1. Parameter that is associated with enabling/disabling vehicle topics
2. Don't modify external parameters
3. Add documentation on external parameter configuration to get this to work

AlexKlimaj committed:
dronecan: don't init if can interface fails to init

AlexKlimaj committed:
Update GPS submodule

MaEtUgR committed:
system_params: clarify SYS_HAS_MAG description

after answering the questions:
Ah the value can be 2? Should I set the number of magnetometers the board has or include the external ones?

mahimayoga committed:
FW land detector: do not check for groundspeed if invalid.

In case the local position speed estimate is not valid, it is assumed to be 0m/s and
thus check always passes.

mahimayoga committed:
FW land detector: Introduce max rotational speed condition (new param: LNDFW_ROT_MAX).

Checks that the filtered norm of the angular velocity is below LNDFW_ROT_MAX.

mahimayoga committed:
MC land detector: fix parameter description for LNDMC_ROT_MAX

PetervdPerk-NXP committed:
Tropic-community use flash as storage using LittleFS (#24158)

 * Update NuttX

sfuhrer committed:
drivers: ulanding: description: fix link to user guide section (#24189)

Signed-off-by: Silvan <silvan@auterion.com>

mbjd committed:
SIH: Add Standard VTOL Airframe (#24175)

* add standard vtol airframe to SIH.

mostly took changes from 4d930bde and applied to main.

generate_fw_aerodynamics now takes four arguments rather than using the
_u class member, because depending on vehicle type _u is used
differently.

RomanBapst committed:
Integrator: use 32bit integer to store dt to avoid overflow

Signed-off-by: RomanBapst <bapstroman@gmail.com>

RomanBapst committed:
VehicleImu: use 32 bit integer for dt

Signed-off-by: RomanBapst <bapstroman@gmail.com>

Jaeyoung-Lim committed:
Remove circular dependency if control allocation (#24195)

... and actuator effectiveness

Jaeyoung-Lim committed:
Take 2: Cleanup circular dependencies ActuatorEffectiveness

AlexKlimaj committed:
boards: disable multi ekf on all ark flight controllers

RomanBapst committed:
VehicleOpticalFlow: use 32bit integer for dt

Signed-off-by: RomanBapst <bapstroman@gmail.com>

mahimayoga committed:
MC-stabilized: rescale thrust input to hover thrust at zero stick input

Use hover thrust estimate in stabilized mode to rescale stick inputs. Prevents vehicle from losing/gaining altitude when switching from position to stabilized mode.

mahimayoga committed:
MC-hte: use allocated thrust as input for hover thrust estimator.

Improves estimates on vehicles where thrust is often saturating.

sfuhrer committed:
Enable CONFIG_EKF2_AUX_GLOBAL_POSITION by default

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

sfuhrer committed:
boards: disable CONFIG_EKF2_AUX_GLOBAL_POSITION on some boards

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

dakejahl committed:
logger: always log can_interface_status (#24071)

* logger: always log can_interface_status

* logger: log topic can_interface_status using CONFIG_BOARD_UAVCAN_INTERFACES

MaEtUgR committed:
NuttX: update submodule to branch px4_firmware_nuttx-10.3.0+ with "FlexSPI allow RWW" merged

MaEtUgR committed:
Revert "FLightTaskAuto: limit nudging speed based on distance sensor"

This reverts commit 97cb933cff6d837e59dd55123804cd78abdca22a.

bresch committed:
ekf2: constrain max variance by zero innovation update

Clipping the variance of the covariance matrix has a destabilizing
effect as it increases the correlation between the states.

Commit:
boards configs: add airspeed driver for micoair743 aio and v2

PX4BuildBot committed:
Update submodule mavlink to latest Wed Jan 15 00:39:11 UTC 2025

    - mavlink in PX4/Firmware (fd5b52d4c53f35a520646a6c4ec75588f6b87e0f): https://github.com/mavlink/mavlink/commit/5e3a42b8f3f53038f2779f9f69bd64767b913bb8
    - mavlink current upstream: https://github.com/mavlink/mavlink/commit/619947d8bc29e80eecff18b0f4fecc42d9e171dd
    - Changes: https://github.com/mavlink/mavlink/compare/5e3a42b8f3f53038f2779f9f69bd64767b913bb8...619947d8bc29e80eecff18b0f4fecc42d9e171dd

    619947d8 2024-12-19 Hamish Willee - common.xml - PING fix (#2197)
2f44ceff 2024-12-18 Julian Oes - common: use camera ID for CAMERA_IMAGE_CAPTURED (#2196)

Jaeyoung-Lim committed:
Make control allocation and actuator effectiveness a non-module-specific library (#24196)

* Remove more circular dependencies with ActuatorEffectiveness

* Separate vehicle specific actuator effectiveness

Keep actuator effectivenss in control allocator

* Remove test dependency for now

* Group library directories
Fix

* Change directory names

* Rebase fix

bresch committed:
[Multirotor] add yaw torque low pass filter (#24173)


co-authored-by: danielmellinger <107884356+danielmellinger@users.noreply.github.com>
co-authored-by: Eric Katzfey <eric.katzfey@modalai.com>

MaEtUgR committed:
Control allocation: make heli rpm control an optional build flag disabled by default

to save flash.
The rpm capture dirver is also disabled on default releases

bresch committed:
ekf2: do not auto-generate sideslip measurement jacobian

This is to trade a bit of CPU load for more flash space.

MaEtUgR committed:
rc_update params: remove comment about default static mixing

Was added back in 2014:
7441efde4745c0dddc08a36a0bbf83307f82948a

bresch committed:
ekf2: allow sideslip fusion to always start with airspeed fusion

This allow sideslip fusion to start during VTOL front transition or even
on a multirotor with a vertical stabilizer and an airspeed sensor for example.

mbjd committed:
stream ATTITUDE_QUATERNION in low bandwidth mode

dagar committed:
ekf2: yaw estimator additional validity checks

mahimayoga committed:
collision-prevention: extract rotation scaling function into new ObstacleMath library.

New library created for static and/or repeated code across collision prevention and driver files.

mahimayoga committed:
obstacle-math: add unit tests for project_distance_on_horizontal_plane function.

mahimayoga committed:
sf45: scale measured distance with pitch and roll.

Calls function from ObstacleMath library that accounts for the vehicle's attitude w.r.t the obstacle. Obstacles are assumed to be flat, vertical walls.

Pedro-Roque committed:
Control Allocator: Add option for metric allocation (skip normalization) (#24199)

* add: metric allocation

* add: actual files

* rft: moved metric allocation to pseudo-inverse via flag with public method

* del: removed metric allocation test and added test in pseudo-inverse testing

* rft: deleted extra newline at the end of pseudo inverse test file

* feat: removed unnecessary log include

bresch committed:
VTOL params: fix param name

haumarco committed:
Fix max-hagl restriction to position/altitude control (#23667)

* fix max-hagl restriction to position/altitude control

* max hagl vel restriction in ManAcc position mode

* use interpolate func, change naming

* simplyfied vertical vel limitation

* move velocity-constraint adjustment to StickAccelXY

haumarco committed:
increase EKF2_RNG_FOG for FW and VTOL

RomanBapst committed:
vtol_takeoff: store altitude on takeoff and don't use home position altitude
as vehicle does not need to be close to home position

Signed-off-by: RomanBapst <bapstroman@gmail.com>

AlexKlimaj committed:
boards: ark fpv add gimbal module (#24229)

dakejahl committed:
gz: increase timeout for service request (#24164)

* gz: increase timeout for service request

* change error messages to warnings, specify retrying

* fix typo

julianoes committed:
Tools: remove unused/wrong define and fix comment

dakejahl committed:
gz: remove model spawn offset (#24165)

dakejahl committed:
uavcan: fix hw_errors from mutex lock, hide ESC/Servo status if no function set (#23888)

AlexKlimaj committed:
dshot: remove dshot 1200

PavloZMN committed:
Optical Flow: add unit testes for only using downward distance sensor (#23266)

* Test for Optical Flow checks correct camera position

* Formatting fixed

* Update src/modules/sensors/vehicle_optical_flow/test/CMakeLists.txt

Co-authored-by: Mathieu Bresciani <brescianimathieu@gmail.com>

* Update src/modules/sensors/vehicle_optical_flow/test/VehicleOpticalFlowTest.cpp

Co-authored-by: Mathieu Bresciani <brescianimathieu@gmail.com>

* Update src/modules/sensors/vehicle_optical_flow/test/VehicleOpticalFlowTest.cpp

Co-authored-by: Silvan Fuhrer <silvan@auterion.com>

* For test GIVEN/WHEN/THEN added

---------

Co-authored-by: Mathieu Bresciani <brescianimathieu@gmail.com>
Co-authored-by: Silvan Fuhrer <silvan@auterion.com>

PX4BuildBot committed:
update all px4board kconfig

PX4BuildBot committed:
boards: update all NuttX defconfigs

RomanBapst committed:
Mission feasibility checks: make adding new check less error prone (#24241)

* make adding new feasibility checks less prone to errors

Signed-off-by: RomanBapst <bapstroman@gmail.com>

* Update src/modules/navigator/MissionFeasibility/FeasibilityChecker.hpp

Co-authored-by: Silvan Fuhrer <silvan@auterion.com>

---------

Signed-off-by: RomanBapst <bapstroman@gmail.com>
Co-authored-by: Silvan Fuhrer <silvan@auterion.com>

dagar committed:
rcS: shift sensors + EKF startup earlier

 - we want the drivers, sensors hub, and estimator running as soon as possible to initialize and avoid commander false positives complaining about missing data

AlexKlimaj committed:
drivers/power_monitor/ina238: retry if read fails

* ina238: retry if read fails
* ina238: increase retries and only publish not connected if register check fails
* ina238: use I2C resets

cuav-liu1 committed:
boards: fix 7-nano pwm voltage control pin not initialized

sfuhrer committed:
FW defaults: remove EKF2_GPS_CHECK custom FW setting

We want to align the default over all vehicle types for
this param. There are still some thresholds that are
increased for FW.

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

bresch committed:
ekf-replay: fix airspeed replay

If available, the EKF uses airspeed_validated, not airspeed
